luskan/src/test/java/org/opencadc/luskan/InventoryFunctionConverterTest.java

```diff
@@ -0,0 +1,220 @@
+import ca.nrc.cadc.util.MultiValuedProperties;
+import ca.nrc.cadc.uws.Parameter;
+
+public class InventoryFunctionConverterTest {
+    private static final Logger log = Logger.getLogger(InventoryFunctionConverterTest.class);
+
+    static {
+        Log4jInit.setLevel("org.opencadc.luskan", Level.INFO);
+        Log4jInit.setLevel("net.sf.jsqlparser", Level.INFO);
+    }
+
+    private static final TapSchema tapSchema = TestUtil.mockTapSchema();
+
+    @Test
+    public void testSelect() {
+        String query = "select *, num_copies() from inventory.Artifact where id = '{some artifact id}'";
+        String expected = "SELECT inventory.Artifact.id, inventory.Artifact.contentLength, cardinality(inventory.Artifact.siteLocations) FROM inventory.Artifact WHERE id = '{some artifact id}'";
+        doTest(query, expected, "inventory.Artifact.siteLocations");
+    }
+
+    @Test
+    public void testWhere() {
+        String query = "select count(*) from inventory.Artifact where num_copies() = 3";
+        String expected = "select count(*) from inventory.Artifact where cardinality(inventory.Artifact.siteLocations) = 3";
+        doTest(query, expected, "inventory.Artifact.siteLocations");
+    }
+
+    @Test
+    public void testGroupByAs() {
+        String query = "select count(*), num_copies() as num from inventory.Artifact group by num";
+        String expected = "select count(*), cardinality(inventory.Artifact.siteLocations) as num from inventory.Artifact group by num";
+        doTest(query, expected, "inventory.Artifact.siteLocations");
+    }
+
+    @Test
+    public void testAlias() {
+        String query = "select count(*), num_copies() from inventory.Artifact as a";
+        String expected = "select count(*), cardinality(a.siteLocations) from inventory.Artifact as a";
+        doTest(query, expected, "a.siteLocations");
+    }
+
+    @Test
+    public void testMultipleArtifactTables() {
+        String query = "select count(*), num_copies() from inventory.Artifact as a, temp.Artifact as b where a.id = b.id";
+        String expected = "select count(*), cardinality(a.siteLocations) from inventory.Artifact as a, temp.Artifact as b where a.id = b.id";
+        doTest(query, expected, null);
+    }
+
+    @Test
+    public void testNoFromTables() {
+        String query = "select num_copies()";
+        doTestFailure(query);
+    }
+
+    @Test
+    public void testNotInventorySchema() {
+        String query = "select count(*), num_copies() from temp.Artifact";
+        doTestFailure(query);
+    }
+
+    @Test
+    public void testMultipleInventoryArtifactTables() {
+        String query = "select count(*), num_copies() from inventory.Artifact as a, inventory.Artifact as b";
+        doTestFailure(query);
+    }
+
+    @Test
+    public void testNoInventoryArtifactTable() {
+        String query = "select num_copies() from DeletedArtifactEvent";
+        doTestFailure(query);
+    }
+
+    @Test
+    public void test() {
```

Comment:
Is this supposed to fail because it is `from DeletedArtifactEvent` or because `num_copies()` is in the group by? The method name doesn't express intent and it could fail for 3 different reasons and thus not be testing what it's supposed to (3rd: the mock tap_schema probably includes inventory.DeletedArtifactEvent).

Observe this:
```
cadctest=> select count(*),cardinality(siteLocations) from inventory.Artifact group by cardinality(siteLocations);
 count | cardinality 
-------+-------------
     3 |            
```
which says there are 3 artifacts with null siteLocations... so function can appear in group by without an alias and should be converted there. This test appears to be a duplicate "wrong table" test.

So, please add a `testGroupBy` (without alias) and verify that both uses of num_copies() get converted. This `test()` doesn't seem to test anything new.

Reply:
The jsqlparser will not parse a query with a function name in the group by.
`select count(*), num_copies() from inventory.Artifact group by num_copies()`
will throw a JSQLParserException on `()` in `group by num_copies()` 

The test wasn't named because it was a throw away to confirm a function can not be used in group by, and should have been deleted before the commit.